crossdc-producer/src/main/java/org/apache/solr/update/processor/MirroringUpdateProcessor.java

```diff
@@ -107,18 +110,24 @@ UpdateRequest createMirrorRequest() {
     doc.removeField(CommonParams.VERSION_FIELD); // strip internal doc version
     final long estimatedDocSizeInBytes = ObjectSizeEstimator.estimate(doc);
     log.info("estimated doc size is {} bytes, max size is {}", estimatedDocSizeInBytes, maxMirroringDocSizeBytes);
+    producerMirroringMetrics.getDocumentsSize().update(estimatedDocSizeInBytes);
     final boolean tooLargeForKafka = estimatedDocSizeInBytes > maxMirroringDocSizeBytes;
     if (tooLargeForKafka && !indexUnmirrorableDocs) {
       throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "Update exceeds the doc-size limit and is unmirrorable. id="
 
           + cmd.getPrintableId() + " doc size=" + estimatedDocSizeInBytes + " maxDocSize=" + maxMirroringDocSizeBytes);
     } else if (tooLargeForKafka) {
+      producerMirroringMetrics.getTooLargeDocuments().inc();
       log.warn(
           "Skipping mirroring of doc {} as it exceeds the doc-size limit ({} bytes) and is unmirrorable. doc size={}",
           cmd.getPrintableId(), maxMirroringDocSizeBytes, estimatedDocSizeInBytes);
     }
 
-    super.processAdd(cmd); // let this throw to prevent mirroring invalid reqs
+    try {
+      super.processAdd(cmd); // let this throw to prevent mirroring invalid reqs
+    } catch (IOException exception) {
+      producerMirroringMetrics.getSavedDocuments().inc();
```

Comment:
This should re-throw, that's the point of calling it first so that we avoid sending documents that will likely cause the same error after mirroring.
Also, "savedDocuments" seems to be wrong, they are likely not "saved" locally because we got an exception. Maybe "errors.local" ?

Reply:
Fixed to `local`, exception will be thrown to process it correctly.